tests/facility/settings/activityDefinitions/activityDefinitionList.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { expect, test } from "@playwright/test";
+
+import { getFacilityId } from "tests/support/facilityId";
+
+test.use({ storageState: "tests/.auth/user.json" });
+
+let facilityId: string;
+let categorySlug: string;
+
+test.beforeAll(() => {
+  facilityId = getFacilityId();
+  categorySlug = `f-${facilityId}-lab-tests-activity_definition`;
+});
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(
+    `/facility/${facilityId}/settings/activity_definitions/categories/${categorySlug}`,
+  );
+});
+
+test.describe("activity definition list", () => {
+  test("should display activity definitions list with correct initial state", async ({
+    page,
+  }) => {
+    await expect(
+      page.locator('[data-slot="breadcrumb-link"]').filter({
+        hasText: /activity definition/i,
+      }),
+    ).toBeVisible();
+    await expect(
+      page.locator('[data-slot="breadcrumb-page"]').filter({
+        hasText: /lab tests/i,
+      }),
+    ).toBeVisible();
+
+    await expect(
+      page
+        .locator('[data-slot="select-trigger"]')
+        .filter({ hasText: /status/i })
+        .filter({ hasText: /active/i }),
+    ).toBeVisible();
+
+    const tableRows = page.locator(
+      '[data-slot="table-body"] [data-slot="table-row"]',
+    );
+    await expect(tableRows.nth(3)).toBeAttached();
+    const rowCount = await tableRows.count();
+    expect(rowCount).toBeGreaterThanOrEqual(4);
+
+    await expect(
+      page.getByText(/showing \d+ of \d+ activity definitions/i),
+    ).toBeVisible();
+
+    await page.getByRole("link", { name: /view/i }).first().click();
+
+    await expect(page).toHaveURL(
+      new RegExp(`/facility/${facilityId}/settings/activity_definitions/.*`),
+    );
+
+    await page.getByRole("button", { name: /back/i }).click();
+
+    await expect(page).toHaveURL(
+      new RegExp(
+        `/facility/${facilityId}/settings/activity_definitions/categories/${categorySlug}`,
+      ),
+    );
+
+    await page.getByRole("link", { name: /edit/i }).first().click();
+
+    await expect(page).toHaveURL(
+      new RegExp(
+        `/facility/${facilityId}/settings/activity_definitions/.*/edit`,
+      ),
+    );
+  });
+
+  test("should filter activity definitions by search", async ({ page }) => {
+    const searchInput = page.getByPlaceholder(/search activity definitions/i);
+    await searchInput.fill("Lipid Panel");
+
+    const tableRows = page.locator(
+      '[data-slot="table-body"] [data-slot="table-row"]',
+    );
+    await expect(tableRows).toHaveCount(1);
+    await expect(tableRows.filter({ hasText: "Lipid Panel" })).toBeVisible();
+
+    await expect(
+      page.getByText(/showing 1 of 1 activity definitions/i),
+    ).toBeVisible();
+  });
+
+  test("should filter by status and clear filter", async ({ page }) => {
```

Comment:
- verify the status on each activity definition by checking some definition are coming